test/org/sagebionetworks/bridge/services/ExternalIdServiceTest.java

```diff
@@ -156,4 +148,150 @@ public void deleteExternalIdsWithValidationEnabled() {
+        
+        // Not validated, set as a null option
+        verifySetAsOption(null);
+    }
+    
+    @Test
+    public void updateExternalIdValidatedWithSameValue() {
+        setupExternalIdTest(true, EXT_ID);
+        
+        externalIdService.reserveExternalId(STUDY, EXT_ID, HEALTH_CODE);
+        externalIdService.assignExternalId(STUDY, EXT_ID, HEALTH_CODE);
+        
+        // Submitting same value again with validation just resets option (noop)
+        verifySetAsOption(EXT_ID);
+    }
+
+    @Test
+    public void updateExternalIdNotValidatedWithSameValue() {
+        setupExternalIdTest(false, EXT_ID);
+        
+        externalIdService.reserveExternalId(STUDY, EXT_ID, HEALTH_CODE);
+        externalIdService.assignExternalId(STUDY, EXT_ID, HEALTH_CODE);
+        
+        // Submitting same value again without validation saves same option value (noop)
+        verifySetAsOption(EXT_ID);
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void updateExternalIdValidatedWithChangedValue() {
+        setupExternalIdTest(true, EXT_ID);
+        
+        // Updating a validated ID throws an exception
+        externalIdService.reserveExternalId(STUDY, "newExternalId", HEALTH_CODE);
+        externalIdService.assignExternalId(STUDY, "newExternalId", HEALTH_CODE);
+    }
+
+    @Test
+    public void updateExternalIdNotValidatedWithChangedValue() {
+        setupExternalIdTest(false, EXT_ID);
+        
+        externalIdService.reserveExternalId(STUDY, "newExternalId", HEALTH_CODE);
+        externalIdService.assignExternalId(STUDY, "newExternalId", HEALTH_CODE);
+
+        // Updating a non-validated ID sets it as the new option
+        verifySetAsOption("newExternalId");
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void updateExternalIdValidatedNoValue() {
+        setupExternalIdTest(true, EXT_ID);
+        
+        // Nulling a validated value throws an exception
+        externalIdService.reserveExternalId(STUDY, null, HEALTH_CODE);
+        externalIdService.assignExternalId(STUDY, null, HEALTH_CODE);
+    }
+
+    @Test
+    public void updateExternalIdNotValidatedNoValue() {
+        setupExternalIdTest(false, EXT_ID);
+        
+        externalIdService.reserveExternalId(STUDY, null, HEALTH_CODE);
+        externalIdService.assignExternalId(STUDY, null, HEALTH_CODE);
+        
+        // Nulling a non-validated value sets the option to null
+        verifySetAsOption(null);
+    }
+    
+    @Test(expected = BadRequestException.class)
+    public void updateExternalIdValidatedNewValue() {
+        setupExternalIdTest(true, EXT_ID);
+        
+        externalIdService.reserveExternalId(STUDY, "newExternalId", HEALTH_CODE);
+        externalIdService.assignExternalId(STUDY, "newExternalId", HEALTH_CODE);
+        
+        // Updating validated value with new value throws exception
+        verifySetAsReservation("newExternalId");
+    }
```

Comment:
Removed the four redundant tests. I would like to leave the remaining 10 tests for the various combinations of calls with existing states.